### ISSUE TITLE
ctc loss的改进建议

### ISSUE THREAD

**Issue by liwenju0** [Topdu/OpenOCR#20]
if self.use_focal_loss:
      # 使用 torch.clamp 限制 loss 的范围，避免指数计算时溢出
      clamped_loss = torch.clamp(loss, min=-20, max=20)
      weight = 1 - torch.exp(-clamped_loss)
      weight = torch.square(weight)
      # 使用 torch.where 避免乘以零权重
      loss = torch.where(weight > 0, loss * weight, loss)
  loss = loss.mean()
  
实际使用时，打开focal loss，这里容易出现nan

**Comment by Topdu** [comment]
Thank you for your suggestion. Would you be willing to create a PR? Once the PR is merged in, you can be highlighted as a contributor.

**Comment by liwenju0** [comment]
> Thank you for your suggestion. Would you be willing to create a PR? Once the PR is merged in, you can be highlighted as a contributor.

https://github.com/Topdu/OpenOCR/pull/21
